### PR TITLE
Petr895 patch 1

### LINKED ISSUE #141: Article: Portability

### PR DESCRIPTION
**Pull request by Petr895**
Closes #141 

## If an article related PR

- [x] I have checked that my suggestion follows the Contribution guide

- [x]  I have added which issue this PR closes

- [ ] I have 0 warnings from Danger.js

- [x] If I have added images, they are in /files/ folder

## If other PR

- [x] I have added which issue this PR closes

### CODE REVIEW

**Comment by Petr895** [comment]
New article – Software Portability

File: practices/software_portability.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+### Application Portability
+
+Portable applications can be converted from one computer environment to another one. 
+With regards to Windows apps, such programs would not use the Registry. In a totally portable application, all related files would be stored in a single folder so it can be copied with one command. Such a folder has many sub-folders, and they are copied in the same transaction. 
+[Walled gardens](https://www.pcmag.com/encyclopedia/term/walled-garden) prevent data from being easily transferred. 
+
+###  Source Code Portability
+
+Software can be compiled and linked from source code for different operating systems and processors if written in a programming language supporting compilation for the platforms.
```
Comment (ridlees) [inline practices/software_portability.md:47]: Software can be compiled and linked from source code for different operating systems and processors if written in a programming language supporting compilation for the platforms. This is usually a task for the program developers; typical users have neither access to the source code nor the required skills.

Please, rewrite this part

File: practices/software_portability.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+## Strategies for Portability
+
+-   Transferring installed program files to another computer of basically the same architecture.
+-   Reinstalling a program from distribution files on another computer of basically the same architecture.
+-   Building executable programs for different platforms from source code; this is usually called “porting”. 
+
+## Cross-platform Software 
+
+It is computer software that is implemented on multiple computing platforms. Cross-platform software may be divided into two types:
```
Comment (ridlees) [inline practices/software_portability.md:62]: (..) is computer software that is implemented on multiple computing platforms. Cross-platform software may be divided into two types; one requires individual building or compilation for each platform that it supports, and the other one can be directly run on any platform without special preparation.

For example, a cross-platform application may run on Microsoft Windows, Linux, and macOS.

Please, consider rewording this part.

**Review by ridlees — CHANGES_REQUESTED** [review]
Hi @Petr895, thanks for the article, I like it. Please, consider rewording the selected parts (it hurts the article if we keep the too similar to originals

File: practices/software_portability.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+# Software Portability
+
+## TL;DR
+
+Portability in high-level computer programming is the usability of the same software in different environments.
+
+## What Is Software Portability?
+
+Software portability is ability to move software from one machine platform to another. It refers to system software or application software that can be recompiled for a different platform or to software that is available for two or more different platforms.
```
[the remark this answers is not in the document]
Comment (Petr895) [reply practices/software_portability.md:27]: Wording changed.

File: practices/software_portability.md (hunk 1 above)
Comment (Petr895) [reply practices/software_portability.md:47]: Wording changed.

File: practices/software_portability.md (hunk 2 above)
Comment (Petr895) [reply practices/software_portability.md]: Wording changed.

File: practices/smart_goals.md [hunk 4]
```diff
@@ -24,6 +24,8 @@ SMART Goals is a tool that can be used to plan and achieve your goals – both p
 
 ![](/files/smart3.png) 
 
+[SMART goals summary *(author)*](smart3.png)
```
Comment (prokopsimek) [inline practices/smart_goals.md:27]: invalid image link

Comment (prokopsimek) [reply practices/smart_goals.md:27]: source should navigate to any website - if it is your image, please navigate e.g. to dxheroes.io

Comment (Petr895) [reply practices/smart_goals.md]: Link changed.

**Review by ridlees — APPROVED** [review]

**Pull request merged**